src/Cache/ApcuCache.php

```diff
@@ -53,31 +55,33 @@ public function remove(string $key): bool
     }
 
     /**
-     * Retrieve an item from the cache.
+     * Internal method to retrieve the raw cache value;
+     * needs to return a Value object or null if not found
      *
-     * @param  string  $key
+     * @param string $key
      * @return mixed
      */
-    public function retrieve(string $key)
+    public function retrieve(string $key): ?Value
```

Comment:
Let's be careful with such type hinting. You cannot have your own Value extension and return that in a driver, which I think is pretty limiting.

Reply:
Fixed now. I have also updated the return type hints in the doc blocks.